### DISCUSSION TITLE
Question about server-and-client-components in appDir

### DISCUSSION THREAD

**Discussion by gyhyfj** [vercel/next.js#47244] [Help] [answered]
The beta document told me to do this:

>To improve the performance of your application, we recommend moving Client Components to the leaves of your component tree where possible

So when I write my `page.tsx`, must I write Client Components in other .tsx files, or I can write them divided but still in one .tsx file?

**Comment by icyJoseph** [comment, marked as answer]
It's more like, everything is a server component, unless you place the 'use client' directive at the top of the file, or module rather. 

> A "use client" directive at the top of a file defines that it's a boundary between server and client.

https://github.com/reactjs/rfcs/blob/main/text/0227-server-module-conventions.md#basic-example states:

> When a Component with a "use client" directive (similar to "use strict") is imported in a "React Server" environment its exports gets replaced with a special "Reference" object. This object can't be accessed directly in that file but it can be passed into React as if it was a plain component.
>
> React, together with the bundler, knows how to send this reference to the client. On the client it's rendered as a Client component. The real file never actually gets imported on the server.

So the Client Components bits, need to go on their own file. It is a separation by module kind of pattern.
